### PR TITLE
Added mention of HTML5 fields in the docs

### PR DESCRIPTION
**Pull request by Deathnerd**
Per request of #275, adding the HTML5 fields to the Fields documentation

### CODE REVIEW

**Comment by davidism** [comment]
Looks like you used a different email than the one associated with your GitHub account. If you want to change that, you can use `git commit --author "Name <email>" --amend` then `git push -f`.

**Comment by Deathnerd** [comment]
Thanks for that!

File: docs/fields.rst [hunk 1]
```diff
@@ -319,6 +319,50 @@ refer to a single input from the form.
 
         {{ form.username(size=30, maxlength=50) }}
 
+HTML5 Fields
+------------
+
+In addition to basic HTML fields, WTForms also supplies fields for the HTML5
+standard. These fields can be accessed under the `wtforms.fields.html5` namespace.
```
Comment (ben4ever) [inline docs/fields.rst]: Decorate `` `wtforms.fields.html5` `` as ``:mod:`wtforms.fields.html5` ``.

File: docs/fields.rst [hunk 2]
```diff
@@ -319,6 +319,50 @@ refer to a single input from the form.
...
+In reality, these fields are just convenience fields that extend basic fields
+and implement HTML5 specific widgets. These widgets are located in the `wtforms.widgets.html5`
```
Comment (ben4ever) [inline docs/fields.rst]: Use `:mod:` here as well.

File: docs/fields.rst [hunk 3]
```diff
@@ -319,6 +319,50 @@ refer to a single input from the form.
...
+namespace and can be overridden or modified just like any other widget.
+
+.. autoclass:: SearchField(default field arguments)
+    Represents an `<input type="search">` field
```
Comment (ben4ever) [inline docs/fields.rst]: Use ``` `` ``` instead of `` ` `` to decorate all HTML code.

File: docs/fields.rst [hunk 4]
```diff
@@ -319,6 +319,50 @@ refer to a single input from the form.
...
+    Represents an `<input type="date">` field
+
+.. autoclass:: TimeField(default field arguments, format='%H:%M')
+    Represents an `<input type="time">` field
+
+.. autoclass:: DateTimeLocalField(default field arguments, format='%Y-%m-%d %H:%M:%S')
+    Represents an `<input type="datetime-local">` field
+
+.. autoclass:: IntegerField(default field arguments)
+    Represents an `<input type="number">` field with the `step="1"` attribute
```
Comment (ben4ever) [inline docs/fields.rst]: Also use ``` `` ``` instead of `` ` `` here and below for the HTML `step` attribute.

**Comment by Deathnerd** [comment]
Thanks for the suggestions. I'll clean these up tomorrow and push them up

**Comment by Deathnerd** [comment]
Cool. I'm still learning how Sphinx works. Thanks for the heads up!

**Comment by Deathnerd** [comment]
I'm having some issues with building. Sphinx doesn't seem to like the fact that these classes are in the `html5` module. I would really like to continue using `autoclass` without messing up the `module` scope for the rest of the document. Thoughts?

**Comment by ftm** [comment]
@davidism, what are the changes that still need to be done to get this sorted? This seems like something that really should be in the docs so I'm happy to make the changes myself.

**Comment by davidism** [comment]
Does it build? Looks fine otherwise.

**Comment by ftm** [comment]
It didn't build, it needed a few changes so I made those and merged it in a separate PR #409 but it should also realise this has been merged too.

**Pull request merged**